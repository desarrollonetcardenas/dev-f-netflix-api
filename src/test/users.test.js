const { graphql } = require('graphql');
const { schema } = require('../index');
const setupTest = require('./helpers');

const mutationRegister = `
    mutation Register( $first_name:String!, $last_name:String!, $email:String!, $password:String!, $birth_date: String!){
        signup(data:{ first_name:$first_name,
                        last_name:$last_name,
                        email:$email,
                        password:$password,
                        birth_date:$birth_date
                    }){
                        token
                    }
        }`;

describe("Register user works correctly", () => {
    beforeEach( async() => await setupTest());

    it("Should create user correctly", async() => {
        const first_name = "Prueba";
        const last_name = "Prueba";
        const email = "prueba3@email.com";
        const password = "miprueba";
        const birth_date = "1900-01-01";

        const res = await graphql( schema, mutationRegister, null, {}, { first_name, last_name, email, password, birth_date });
        expect(res).toMatchSnapshot();
        expect(res).toHaveProperty("data");
    });

});
