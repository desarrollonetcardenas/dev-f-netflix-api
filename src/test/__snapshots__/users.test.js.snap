// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register user works correctly Should create user correctly 1`] = `
Object {
  "data": Object {
    "signup": null,
  },
  "errors": Array [
    [GraphQLError: E11000 duplicate key error collection: test.users index: email_1 dup key: { : "prueba3@email.com" }],
  ],
}
`;
